blockstore/api.go

```diff
@@ -31,31 +40,44 @@ func (a *apiBlockstore) DeleteBlock(context.Context, cid.Cid) error {
 }
 
 func (a *apiBlockstore) Has(ctx context.Context, c cid.Cid) (bool, error) {
+	if a.cache.Contains(c) {
+		return true, nil
+	}
 	return a.api.ChainHasObj(ctx, c)
 }
 
 func (a *apiBlockstore) Get(ctx context.Context, c cid.Cid) (blocks.Block, error) {
+	if bb, ok := a.cache.Get(c); ok {
+		return blocks.NewBlockWithCid(bb, c)
+	}
 	bb, err := a.api.ChainReadObj(ctx, c)
 	if err != nil {
 		return nil, err
 	}
+	a.cache.Add(c, bb)
 	return blocks.NewBlockWithCid(bb, c)
 }
 
 func (a *apiBlockstore) GetSize(ctx context.Context, c cid.Cid) (int, error) {
+	if bb, ok := a.cache.Get(c); ok {
```

Comment:
Double checking that we do mean to call `Get` instead of `Peek`, i.e., should the calls to `GetSize` affect cache eviction?

Reply:
eh, probably a good thing to do but I bet if I made this method panic then nothing would break